From 55eb19200d650ead73139ee8444db9119718fd31 Mon Sep 17 00:00:00 2001
From: Sergey Zhuravlevich <zhurxx@gmail.com>
Date: Tue, 12 May 2020 14:23:15 +0200
Subject: [PATCH 02/49] MLK-24065-1: drm: bridge: cadence: fix dp_aux_transfer
 write return value

After exiting the loop in DP_AUX_NATIVE_WRITE it was returning 0. It's supposed
to return the number of bytes transferred on success.

Signed-off-by: Sergey Zhuravlevich <zhurxx@gmail.com>
Acked-by: Sandor Yu <sandor.yu@nxp.com>
Tested-By: Sandor Yu <sandor.yu@nxp.com>
---
 drivers/gpu/drm/bridge/cadence/cdns-dp-core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/bridge/cadence/cdns-dp-core.c b/drivers/gpu/drm/bridge/cadence/cdns-dp-core.c
index acb5c860da73..aa92029f44e9 100644
--- a/drivers/gpu/drm/bridge/cadence/cdns-dp-core.c
+++ b/drivers/gpu/drm/bridge/cadence/cdns-dp-core.c
@@ -67,6 +67,8 @@ static ssize_t dp_aux_transfer(struct drm_dp_aux *aux,
 
 			return ret;
 		}
+		msg->reply = DP_AUX_NATIVE_REPLY_ACK;
+		return msg->size;
 	}
 
 	if (msg->request == DP_AUX_NATIVE_READ) {
-- 
2.29.2

